### PR TITLE
Update CI pipeline to support installing build-tools from npm

### PR DESCRIPTION
**Pull request by tylerbutler**
We use our homegrown build-tools to do some tasks in our CI pipelines. It's helpful to be able to install the local in-repo version of the tools in most cases, but sometimes we want to install from npm instead. This change enables us to control the version of the build tools that are installed via a pipeline variable that's set when the pipeline is run.

### CODE REVIEW

File: build-tools/packages/build-cli/src/commands/generate/buildVersion.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export default class GenerateBuildVersionCommand extends BaseCommand<
         }),
         release: Flags.string({
             description: "Indicates the build is a release build.",
-            options: ["release"],
+            options: ["release", "none"],
```
Comment (tylerbutler) [inline build-tools/packages/build-cli/src/commands/generate/buildVersion.ts:36]: This is being split out into #11683.

File: tools/pipelines/build-azure-local-service.yml [hunk 2]
```diff
@@ -21,6 +21,10 @@ parameters:
     - default
     - skip
     - force
+- name: buildToolsVersionToInstall
+  displayName: Fluid build tools version (default = installs version in repo)
```
Comment (curtisman) [inline tools/pipelines/build-azure-local-service.yml:25]: should it default to not repo?  the repo version does add non-trivial time to the pipeline.

**Review by curtisman — APPROVED** [review]

File: tools/pipelines/build-azure-local-service.yml (hunk 2 above)
Comment (tylerbutler) [reply tools/pipelines/build-azure-local-service.yml:25]: Once the build tools are baked, I'll update the default for main. In lts it'll default to npm, but main for now will use the in-repo version.

File: tools/pipelines/templates/include-set-package-version.yml [hunk 3]
```diff
@@ -12,32 +12,53 @@ parameters:
...
+    displayName: Install Fluid Build Tools (from repo)
+    inputs:
+      targetType: 'inline'
+      workingDirectory: ${{ parameters.buildDirectory }}
+      script: |
+        pushd "$(Build.SourcesDirectory)/build-tools"
+        npm ci
+        popd
+
+- ${{ if ne(parameters.buildToolsVersionToInstall, 'repo') }}:
```
Comment (alexvy86) [inline tools/pipelines/templates/include-set-package-version.yml:45]: Curious if we use an `if-equal` above and an independent `if-not-equal` here over an `else` for some reason?

**Pull request merged**